src/collections/Node.js

```diff
@@ -32,6 +35,9 @@ var traversalMethods = {
    * @return {Collection}
    */
   find: function(type, filter) {
+    if (type.__IS_JSX__) {
```

Comment:
If this was changed to

```diff
+    if (!filter && !types[type]) {
+      filter = type;
+      type = filter.type || 'Node';
+    }
```

instead and move the `matchJSX`, then we don't need special treatment for JSX.